### PR TITLE
Mege SLE-12 into master branch

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: test/lan_udev_auto_test.rb [hunk 1]
```diff
@@ -108,3 +109,26 @@
...
+    udev_rules = [
+      {
+        "rule"  => attr_addr,
+        "value" => value,
+        "name"  => name
+      }
+    ]
+
+    expect(Yast::LanUdevAuto.Import({ "net-udev" => udev_rules }))
+      .to be true
```
Comment (ancorgs) [inline test/lan_udev_auto_test.rb]: better `eq` than `be`

File: test/lan_udev_auto_test.rb [hunk 2]
```diff
@@ -6,105 +6,135 @@
...
-      expect(Yast::LanUdevAuto.getDeviceName(NEW_STYLE_NAME)).to be_equal NEW_STYLE_NAME
+  describe ".getDeviceName" do
+    subject(:getDeviceName) { Yast::LanUdevAuto.getDeviceName(oldname) }
+
+    let(:new_style_name) { "spec0" }
+    let(:mac) { "00:11:22:33:44:FF" }
+    let(:busid) { "0000:00:19.0" }
+
+    # general mocking stuff is placed here
+    before(:each) do
```
Comment (jreidinger) [inline test/lan_udev_auto_test.rb]: `before(:each)` can be replaced by simple `before`

File: src/modules/LanUdevAuto.rb [hunk 3]
```diff
@@ -180,7 +180,7 @@ def Import(settings)
           createUdevFromIfaceName(Ops.get_list(settings, "interfaces", []))
         )
       else
-        @udev_rules = Ops.get_list(settings, "net-udev", [])
+        @udev_rules = settings["net-udev"] || []
```
Comment (jreidinger) [inline src/modules/LanUdevAuto.rb]: I do not get how it fixes internal error. For me it looks equivalent or I overlook something?

Comment (mchf) [inline src/modules/LanUdevAuto.rb]: actual fix is in the other file in the commit. I was also checking if an expected default value is set properly everywhere, so I turned it from converted code into ruby here.

Comment (mchf) [inline src/modules/LanUdevAuto.rb]: note that the fix was already present in the master, so here it disappears in the merge.

File: test/lan_udev_auto_test.rb (hunk 2 above)
Comment (mchf) [inline test/lan_udev_auto_test.rb]: can be, but I prefer to be explicit. Is it blocker?

File: src/modules/LanUdevAuto.rb (hunk 3 above)
Comment (jreidinger) [inline src/modules/LanUdevAuto.rb]: I see only change here and in test suite, spec and changes, nothing else.

Comment (jreidinger) [inline src/modules/LanUdevAuto.rb]: ahm having it in master, exaplains it

File: test/lan_udev_auto_test.rb (hunk 2 above)
Comment (jreidinger) [inline test/lan_udev_auto_test.rb]: not blocker just `nth`, as before(:all) is almost not used, so each is good and intuitive default

Comment (kobliha) [inline test/lan_udev_auto_test.rb]: I use it quite often and nobody ever objected :)

**Comment by jreidinger** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for reviews

**Pull request merged**